kqueen/engines/openstack_kubespray.py

```diff
@@ -620,9 +630,12 @@ def validate_ip(address):
             self.meta["image"] = self.c.get_image(self.cluster.metadata["image_name"])
             if self.meta["image"] is None:
                 raise ValueError("Image '%s' is not found" % self.cluster.metadata["image_name"])
-            self.meta["flavor"] = self.c.get_flavor(self.cluster.metadata["flavor"])
-            if self.meta["flavor"] is None:
-                raise ValueError("Flavor '%s' is not found" % self.cluster.metadata["flavor"])
+            self.meta["master_flavor"] = self.c.get_flavor(self.cluster.metadata["master_flavor"])
```

Comment:
btw, why some checks are not in try-catch block?

Reply:
they are optional